### ISSUE TITLE
Rule no-extra-parens needs an option to allow `return (foo = bar)`

### ISSUE THREAD

**Issue by domenic** [eslint/eslint#6036]
**What version of ESLint are you using?**: 2.9.0

**What parser (default, Babel-ESLint, etc.) are you using?**: default

**Please show your full configuration:** https://gist.github.com/domenic/f6f5b7c294b610e7184b29d5e589d73b

**What did you do? Please include the actual source code causing the issue.**

```js
  parentNode() {
    let node = this._currentNode;
    while (!isNull(node) && node !== this.root) {
      node = node.parentNode;

      if (!isNull(node) && this._filterNode(node) === FILTER_ACCEPT) {
        return (this._currentNode = node);  // THIS LINE, THIS ONE RIGHT HERE
      }
    }
    return null;
  }
````

**What did you expect to happen?** Since I have configured no-return-assign to be "except-parens", I was hoping "no-extra-parens" would have an option that's compatible with my no-return-assign configuration, and would allow the extra parens in that particular case.

**What actually happened? Please include the actual, raw output from ESLint.**

```
C:\Users\Domenic\Dropbox\GitHub\tmpvar\jsdom\lib\jsdom\living\traversal\TreeWalker-impl.js
   46:16  error  Gratuitous parentheses around expression  no-extra-parens
```

This is a request to add another option to no-extra-parens to allow it in the return-assign case.

**Comment by mysticatea** [comment]
Thank you for this issue.

`no-extra-parens` must have the option to avoid conflicting with `no-return-assign`.
`no-extra-parens` has the option to avoid conflicting with `no-cond-assign` currently.

**Comment by kaicataldo** [comment]
I can take a look at this if you're not already, @mysticatea

**Comment by mysticatea** [comment]
@kaicataldo Thank you! I have not started to work on this yet. Yes, `conditionalAssign`-like option for `return` statements.

**Comment by kaicataldo** [comment]
PR created
